pkg/templates/externalccm/ccm.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalccm
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/kubermatic/kubeone/pkg/config"
+	"github.com/kubermatic/kubeone/pkg/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	dynclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Ensure external CCM deployen if Provider.External
+func Ensure(ctx *util.Context) error {
+	if !ctx.Cluster.Provider.External {
+		return nil
+	}
+
+	ctx.Logger.Info("Ensure external CCM is up to date")
+
+	switch ctx.Cluster.Provider.Name {
+	case config.ProviderNameHetzner:
+		return ensureHetzner(ctx)
+	case config.ProviderNameDigitalOcean:
+		return ensureDigitalOcean(ctx)
+	default:
+		ctx.Logger.Infof("External CCM for %q not yet supported, skipping", ctx.Cluster.Provider.Name)
```

Comment:
I'd add a message that the user should deploy it manually, e.g. `External CCM for %q not yet supported, please deploy CCM manually`.

Reply:
`please deploy CCM manually` is implied, since we can't install them 🤷‍♂️

Reply:
I prefer being explicit in the user-facing error messages.

Reply:
I don't want to run pipeline again just for "notice"

Reply:
Considering you have to rebase the PR, you can fix this. :wink:

Reply:
too late, already rebased :P